kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSession.scala

```diff
@@ -29,6 +31,17 @@ abstract class KyuubiSession(
     sessionManager: KyuubiSessionManager)
   extends AbstractSession(protocol, user, password, ipAddress, conf, sessionManager) {
 
-  def getSessionEvent: Option[KyuubiSessionEvent]
+  private[kyuubi] lazy val optimizedConf: Map[String, String] = {
+    val confOverlay = sessionManager.sessionConfAdvisor.getConfOverlay(
```

Comment:
In my opinion, maybe we need a new method for batch conf overlay.

Reply:
by defaults, the batch conf overlay could be same with the original one, but it is better to have two seprate methods.

Reply:
```
  /** The returned conf will overwrite the batch session conf */
  default Map<String, String> getBatchConfOverlay(String user, Map<String, String> sessionConf) {
    return getConfOverlay(user, sessionConf);
  }
```

Reply:
+1. there may be some differences.

Reply:
yeah, I considered this. I do not against add a new method for batch. cc @wForget @yaooqinn what do you think ?

Reply:
I am not sure it's a good idea. if so, we shall add APIs for different resource managers, engines, workloads, etc.

Reply:
How about inject the type of request to the `sessionConf` ? For example: `kyuubi.session.type -> batch` then the plugin can catch the key and do something.

Reply:
The interface of `sessionConf` is flexible, user can inject any configs by theirself and catch it inside plugin. So we can leave it as is.

Reply:
> The interface of sessionConf is flexible, user can inject any configs by theirself and catch it inside plugin. So we can leave it as is.

Agree with that.
The sessionConfAdvisor should be developed by kyuubi admin and they can inject some reserved key words for the sessionConfAdvisor.

We can add some comments in the interface.

Reply:
sure, enriched the interface comment